chalk-solve/src/clauses.rs

```diff
@@ -277,6 +293,36 @@ fn program_clauses_that_could_match<I: Interner>(
             let associated_ty_datum = db.associated_ty_data(alias.associated_ty_id);
             let trait_id = associated_ty_datum.trait_id;
             let trait_parameters = db.trait_parameters_from_projection(alias);
+
+            if (alias
```

Comment:
Comment? I think:

// Flounder if both the self-type *and* the target type are unknown.
//
// e.g., `Normalize(<?X as Iterator>::Item = ?Y)`

Reply:
I've removed this one in favor of just the floundering on unknown self type.